freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/repair/changeprimarygw/action/ChangePrimaryGatewayActions.java

```diff
@@ -143,6 +147,27 @@ protected void doExecute(ChangePrimaryGatewayContext context, StackEvent payload
         };
     }
 
+    @Bean(name = "CHANGE_PRIMARY_GATEWAY_HEALTH_CHECK_STATE")
+    public Action<?, ?> healthCheckAction() {
+        return new AbstractChangePrimaryGatewayAction<>(StackEvent.class) {
+            @Override
+            protected void doExecute(ChangePrimaryGatewayContext context, StackEvent payload, Map<Object, Object> variables) {
+                Stack stack = context.getStack();
+
+                Selectable request;
+                if (isFinalChain(variables)) {
+                    stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.REPAIR_IN_PROGRESS, "Checking the health");
+                    request = new HealthCheckRequest(stack.getId(), false);
+                } else {
+                    stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.REPAIR_IN_PROGRESS, "Skipping the health check");
```

Comment:
I find this odd to set the statusreason to `Skipping...` I don't think it would mean anything to the customer. We might just log it here and don't update the status